app/controllers/articles_controller.rb

```diff
@@ -61,9 +61,18 @@ def create
     url = RequestHelper.url_from_param(params.dig(:article, :url))
     title, fetched_url = RequestHelper.extract_title_from_page(url)
 
-    new_article = Article.new(user: current_user, url: fetched_url, title: title)
+    # check if article already exists
+    article = current_user.articles.find_by_url(fetched_url)
 
-    if new_article.save
+    if article
+      article.update(url: fetched_url, title: title, updated_at: Time.now)
+    else
+      new_article = Article.new(user: current_user, url: fetched_url, title: title)
+    end
+
+    if article
+      flash[:success] = 'Looks like you already have this in your list - it\'s now at the top of your list and its expiry was reset'
```

Comment:
I found a simpler sentence structure to improve this:

```suggestion
      flash[:success] = 'This was already in your list, so it's now at the top for easier access. 👇'
```